src/core/interp.c

```diff
@@ -5715,9 +5744,10 @@ void MVM_interp_run(MVMThreadContext *tc, void (*initial_invoke)(MVMThreadContex
                     target->o = check;
                 goto NEXT;
             }
-            OP(sp_rebless):
-                if (!REPR(GET_REG(cur_op, 2).o)->change_type) {
-                    MVM_exception_throw_adhoc(tc, "This REPR cannot change type");
+            OP(sp_rebless): {
+                MVMObject *obj = GET_REG(cur_op, 2).o;
+                if (!REPR(obj)->change_type) {
+                    MVM_exception_throw_adhoc(tc, "REPR %s (%s) cannot change type", REPR(obj)->name, MVM_6model_get_debug_name(tc, obj));
```

Comment:
Why not refactor the rest of the references to obj as well?

Reply:
Hm. Or should I just partially revert and use `GET_REG(cur_op, 2).o` in the arguments to `MVM_exception_throw_adhoc`?